distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Duration.enso

```diff
@@ -262,3 +263,10 @@ type Duration
         if self.milliseconds==0 . not then b.append ["milliseconds", self.milliseconds]
         if self.nanoseconds==0 . not then b.append ["nanoseconds", self.nanoseconds]
         JS_Object.from_pairs b.to_vector
+
+    ## PRIVATE
+       Convert Duration to a friendly string.
+    to_display_text : Text
+    to_display_text self =
+        seconds = self.seconds + self.nanoseconds/1000000000
+        self.hours.to_text + "h" + self.minutes.format "00" + "m" + seconds.format "00.##########"
```

Comment:
This will yield stuff like `0h 00m 12.34567` shouldn't it be `0h 00m 12.34567s`?

Reply:
I think it would be good if we had at least 1 unit test case for these, to make sure that they at least compile (knowing how easy it used be to break `to_display_text` by refactors in the past).

Can we get it? If not in this PR, can we plan a followup?

Reply:
Also ideally, I think it would be good if we can be consistent with `Time_Of_Day` handling of nanoseconds, i.e. if `nanoseconds == 0`, we should have `0h 05m 09s` and not `0h 05m 09.000000000s`. Do the `#` in the format achieve that? I don't know well enough how this format works - would be good to have a test for it as then this knowledge would not be required - we'd have a practical 'proof'.

Reply:
Yes the `#` automatically adds digits as needed (including trimming the `.`).